### PR TITLE
Reduce Verbosity

### PR DESCRIPTION
**Pull request by dylanjm**
--------
Pull Request Description
--------
##### What issue does this change request address?
Continues #125

Depends on idaholab/raven#1757

##### What are the significant changes in functionality due to this change request?

- Remove unused code in Cases.py
- Adds an explicit `<verbosity>` node to `<RunInfo>` block
- Suppress ancillary RAVEN warnings
- Remove print statements and leverage MessageHandler in RAVEN
- Let Template class inherit from Base class. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by dylanjm** [comment]
Here is the previous output of HERON on `tests/integration_tests/workflows/storage/heron_input.xml`
```
Starting HERON ...
 ... RAVEN located at "/Users/mcdodj/Documents/projects/raven"
 ... Activating CONDA environment ...
CONDA
ravenLibs             *  /usr/local/Caskroom/miniconda/base/envs/ravenLibs
Running HERON ...
Loading plugin "HERON" at /Users/mcdodj/Documents/projects/HERON
 ... successfully imported "HERON" ...
Loading plugin "TEAL" at /Users/mcdodj/Documents/projects/TEAL
 ... successfully imported "TEAL" ...
 ... loading economics ...
 ... loading economics ...
 ... loading economics ...
 ... loading economics ...
 ... ... loading cash flow "e_sales"
 ... loading economics ...
 ... ... loading cash flow "e_sales"
(            ) UTILS                    : Message         -> importing module /Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/transfers
(    0.49 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
(    0.49 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
Checking ROM at "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/ARMA/Sine/arma.pk" ...
/usr/local/Caskroom/miniconda/base/envs/ravenLibs/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py:61: FutureWarning: Not all Ray CLI dependencies were found. In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will only be usable via `pip install 'ray[default]'`. Please update your install command.
  "update your install command.", FutureWarning)
/Users/mcdodj/Documents/projects/raven/framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:29: UserWarning: Finding Prescient failed with The 'prescient' distribution was not found and is required by the application
  warnings.warn(f"Finding Prescient failed with {inst}")
For DataGenerator <ARMA> "Price", detected:  interpolated,  clustered.
 -> "Price" interpolates 3 macro steps, and project life is 3, so histories will be trunctated.
Printing simulation state:
  Case:
    name: Sweep_Runs
    mode: sweep
    meric: NPV
    diff_study: None
  Component:
    name: steamer
    Producer:
      produces: ['steam']
      consumes: []
      transfer: None
      capacity: <ValuedParam name: "_capacity" type: "SweepValues">
  Component:
    name: steam_storage
    Storage:
      stores: steam
      rate: None
      capacity: <ValuedParam name: "_capacity" type: "FixedValue">
  Component:
    name: generator
    Producer:
      produces: ['electricity']
      consumes: ['steam']
      transfer: <ValuedParam name: "_transfer" type: "Linear">
      capacity: <ValuedParam name: "_capacity" type: "FixedValue">
  Component:
    name: electr_market
    Demand/Load:
      demands: ['electricity']
      penalty: None
      capacity: <ValuedParam name: "_capacity" type: "FixedValue">
  Component:
    name: electr_flex
    Demand/Load:
      demands: ['electricity']
      penalty: None
      capacity: <ValuedParam name: "_capacity" type: "FixedValue">
  DataGenerator:
    name: Price
    source: %HERON%/tests/integration_tests/ARMA/Sine/arma.pk
    variables: ['Signal']
  DataGenerator:
    name: transfers
    source: transfers.py
    variables: None
HERON: writing files ...
 ... wrote "outer.xml      " to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... wrote "inner.xml      " to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... wrote "cash.xml       " to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... wrote "heron.lib      " to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... wrote "write_inner.py " to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... HERON completed successfully.
```

Here is the updated output of HERON:
```
Starting HERON ...
 ... RAVEN located at "/Users/mcdodj/Documents/projects/raven"
 ... Activating CONDA environment ...
CONDA
ravenLibs             *  /usr/local/Caskroom/miniconda/base/envs/ravenLibs
Running HERON ...
Loading plugin "HERON" at /Users/mcdodj/Documents/projects/HERON
 ... successfully imported "HERON" ...
Loading plugin "TEAL" at /Users/mcdodj/Documents/projects/TEAL
 ... successfully imported "TEAL" ...
(            ) UTILS                    : Message         -> importing module /Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/transfers
(    0.53 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
(    0.53 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
(    0.53 sec) <HERON ARMA>      : Message -> Checking ROM at "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/ARMA/Sine/arma.pk"
(    2.29 sec) <HERON ARMA>      : Message -> For DataGenerator <ARMA> "Price", detected:  interpolated,  clustered.
(    2.31 sec) <Template Driver> : Message -> ========================
(    2.31 sec) <Template Driver> : Message -> HERON: writing files ...
(    2.31 sec) <Template Driver> : Message -> ========================
(    2.31 sec) <Template Driver> : Message -> Wrote "outer.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.31 sec) <Template Driver> : Message -> Wrote "inner.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.31 sec) <Template Driver> : Message -> Wrote "cash.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.32 sec) <Template Driver> : Message -> Wrote "heron.lib" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.32 sec) <Template Driver> : Message -> Wrote "write_inner.py" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... HERON completed successfully.
```

Here is the output when switching to `debug` verbosity on the MessageHandler:
```
Starting HERON ...
 ... RAVEN located at "/Users/mcdodj/Documents/projects/raven"
 ... Activating CONDA environment ...
CONDA
ravenLibs             *  /usr/local/Caskroom/miniconda/base/envs/ravenLibs
Running HERON ...
Loading plugin "HERON" at /Users/mcdodj/Documents/projects/HERON
 ... successfully imported "HERON" ...
Loading plugin "TEAL" at /Users/mcdodj/Documents/projects/TEAL
 ... successfully imported "TEAL" ...
(    0.47 sec) <HERON Case>      : DEBUG   -> Successfully initialized Case Sweep_Runs.
(    0.48 sec) <HERON Component ": DEBUG   -> Loading component "steamer"
(    0.48 sec) <HERON Producer>  : DEBUG   ->  ... loading interaction "produces"
(    0.48 sec) <HERON Component ": DEBUG   -> Loading component "steam_storage"
(    0.48 sec) <HERON Storage>   : DEBUG   ->  ... loading interaction "stores"
(    0.48 sec) <HERON Component ": DEBUG   -> Loading component "generator"
(    0.48 sec) <HERON Producer>  : DEBUG   ->  ... loading interaction "produces"
(    0.49 sec) <HERON Component ": DEBUG   -> Loading component "electr_market"
(    0.49 sec) <HERON Demand>    : DEBUG   ->  ... loading interaction "demands"
(    0.49 sec) <HERON Component ": DEBUG   -> Loading component "electr_flex"
(    0.49 sec) <HERON Demand>    : DEBUG   ->  ... loading interaction "demands"
(            ) UTILS                    : Message         -> importing module /Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/transfers
(    0.49 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
(    0.49 sec) <ValuedParams.Activity.Ac: Message         -> Tracking variable not specified; using "production" ...
(    0.49 sec) <HERON ARMA>      : Message -> Checking ROM at "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/ARMA/Sine/arma.pk"
(    2.05 sec) <HERON ARMA>      : Message -> For DataGenerator <ARMA> "Price", detected:  interpolated,  clustered.
(    2.06 sec) <HERON ARMA>      : DEBUG   -> "Price" interpolates 3 macro steps,and project life is 3, so histories will be trunctated.
(    2.06 sec) <HERON Simulation>: DEBUG   -> ==========================
(    2.06 sec) <HERON Simulation>: DEBUG   -> Printing simulation state:
(    2.06 sec) <HERON Simulation>: DEBUG   -> ==========================
(    2.06 sec) <HERON Case>      : DEBUG   ->   Case:
(    2.06 sec) <HERON Case>      : DEBUG   ->     name: Sweep_Runs
(    2.06 sec) <HERON Case>      : DEBUG   ->     mode: sweep
(    2.06 sec) <HERON Case>      : DEBUG   ->     meric: NPV
(    2.06 sec) <HERON Case>      : DEBUG   ->     diff_study: None
(    2.06 sec) <HERON Component ": DEBUG   ->   Component:
(    2.06 sec) <HERON Component ": DEBUG   ->     name: steamer
(    2.06 sec) <HERON Producer>  : DEBUG   ->     Producer:
(    2.06 sec) <HERON Producer>  : DEBUG   ->       produces: ['steam']
(    2.06 sec) <HERON Producer>  : DEBUG   ->       consumes: []
(    2.06 sec) <HERON Producer>  : DEBUG   ->       transfer: None
(    2.06 sec) <HERON Producer>  : DEBUG   ->       capacity: <ValuedParam name: "_capacity" type: "SweepValues">
(    2.06 sec) <HERON Component ": DEBUG   ->   Component:
(    2.06 sec) <HERON Component ": DEBUG   ->     name: steam_storage
(    2.06 sec) <HERON Storage>   : DEBUG   ->     Storage:
(    2.06 sec) <HERON Storage>   : DEBUG   ->       stores: steam
(    2.06 sec) <HERON Storage>   : DEBUG   ->       rate: None
(    2.06 sec) <HERON Storage>   : DEBUG   ->       capacity: <ValuedParam name: "_capacity" type: "FixedValue">
(    2.06 sec) <HERON Component ": DEBUG   ->   Component:
(    2.06 sec) <HERON Component ": DEBUG   ->     name: generator
(    2.06 sec) <HERON Producer>  : DEBUG   ->     Producer:
(    2.06 sec) <HERON Producer>  : DEBUG   ->       produces: ['electricity']
(    2.06 sec) <HERON Producer>  : DEBUG   ->       consumes: ['steam']
(    2.06 sec) <HERON Producer>  : DEBUG   ->       transfer: <ValuedParam name: "_transfer" type: "Linear">
(    2.06 sec) <HERON Producer>  : DEBUG   ->       capacity: <ValuedParam name: "_capacity" type: "FixedValue">
(    2.06 sec) <HERON Component ": DEBUG   ->   Component:
(    2.06 sec) <HERON Component ": DEBUG   ->     name: electr_market
(    2.06 sec) <HERON Demand>    : DEBUG   ->     Demand/Load:
(    2.06 sec) <HERON Demand>    : DEBUG   ->       demands: ['electricity']
(    2.06 sec) <HERON Demand>    : DEBUG   ->       penalty: None
(    2.06 sec) <HERON Demand>    : DEBUG   ->       capacity: <ValuedParam name: "_capacity" type: "FixedValue">
(    2.06 sec) <HERON Component ": DEBUG   ->   Component:
(    2.06 sec) <HERON Component ": DEBUG   ->     name: electr_flex
(    2.06 sec) <HERON Demand>    : DEBUG   ->     Demand/Load:
(    2.06 sec) <HERON Demand>    : DEBUG   ->       demands: ['electricity']
(    2.06 sec) <HERON Demand>    : DEBUG   ->       penalty: None
(    2.06 sec) <HERON Demand>    : DEBUG   ->       capacity: <ValuedParam name: "_capacity" type: "FixedValue">
(    2.06 sec) <HERON ARMA>      : DEBUG   ->   DataGenerator:
(    2.06 sec) <HERON ARMA>      : DEBUG   ->     name: Price
(    2.06 sec) <HERON ARMA>      : DEBUG   ->     source: %HERON%/tests/integration_tests/ARMA/Sine/arma.pk
(    2.06 sec) <HERON ARMA>      : DEBUG   ->     variables: ['Signal']
(    2.06 sec) <HERON Function>  : DEBUG   ->   DataGenerator:
(    2.06 sec) <HERON Function>  : DEBUG   ->     name: transfers
(    2.06 sec) <HERON Function>  : DEBUG   ->     source: transfers.py
(    2.06 sec) <HERON Function>  : DEBUG   ->     variables: None
(    2.08 sec) <Template Driver> : Message -> ========================
(    2.08 sec) <Template Driver> : Message -> HERON: writing files ...
(    2.08 sec) <Template Driver> : Message -> ========================
(    2.08 sec) <Template Driver> : Message -> Wrote "outer.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.08 sec) <Template Driver> : Message -> Wrote "inner.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.08 sec) <Template Driver> : Message -> Wrote "cash.xml" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.08 sec) <Template Driver> : Message -> Wrote "heron.lib" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
(    2.08 sec) <Template Driver> : Message -> Wrote "write_inner.py" to "/Users/mcdodj/Documents/projects/HERON/tests/integration_tests/workflows/storage/"
 ... HERON completed successfully.
```

File: src/Cases.py [hunk 1]
```diff
@@ -69,6 +69,16 @@ def get_input_specs(cls):
     input_specs.addSub(InputData.parameterInputFactory('mode', contentType=mode_options, strictMode=True,
                                                        descr=desc_mode_options))
 
+    verbosity_options = InputTypes.makeEnumType('VerbosityOptions', 'VerbosityOptionsType',
+                                                ['silent', 'quiet', 'all', 'debug'])
+    desc_verbosity_options = r"""determines the level of verbosity for the outer/inner RAVEN runs. \default{all}.
+                             If ``silent'' only errors are displayed.
+                             If ``quiet'' errors and warnings are displayed.
+                             If ``all'' errors, warnings, and messages are displayed.
+                             If ``debug'' errors, warnings, messages, and debug messages are displayed."""
```
Comment (PaulTalbot-INL) [inline src/Cases.py:78]: do we want a "see RAVEN documentation for more information" caveat here?

File: src/Cases.py [hunk 2]
```diff
@@ -185,6 +195,7 @@ def __init__(self, run_dir, **kwargs):
     self._mode = None           # extrema to find: opt, sweep
     self._metric = 'NPV'        # UNUSED (future work); economic metric to focus on: lcoe, profit, cost
     self.run_dir = run_dir      # location of HERON input file
+    self._verbosity = 'all'     # default verbosity for RAVEN inner/outer
```
Comment (PaulTalbot-INL) [inline src/Cases.py:198]: I suspect we want this on "quiet" perhaps by default? idk

File: src/_utils.py [hunk 3]
```diff
@@ -92,7 +94,10 @@ def get_synthhist_structure(fpath):
   scripts_path = os.path.join(raven_loc, '..', 'scripts')
   sys.path.append(scripts_path)
   from externalROMloader import ravenROMexternal as ravenROM
-  rom = ravenROM(fpath, raven_loc).rom
+  # Why should we get warnings from RAVEN when we are just trying to write an input file.
+  with warnings.catch_warnings():
+    warnings.simplefilter('ignore')
+    rom = ravenROM(fpath, raven_loc).rom
```
Comment (PaulTalbot-INL) [inline src/_utils.py:100]: 💯

File: templates/outer.xml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<Simulation verbosity='debug'>
+<Simulation verbosity='all'>
```
Comment (PaulTalbot-INL) [inline templates/outer.xml:1]: did you change "inner" verbosity" It should probably be "quiet" or "silent" by default. At least "all" will chop the Monte Carlo messages.

**Review by PaulTalbot-INL — COMMENTED** [review]
Some comments for discussion; may not require changes depending on how the conversation goes.

File: src/Cases.py [hunk 5]
```diff
@@ -69,6 +69,16 @@ def get_input_specs(cls):
...
 
+    verbosity_options = InputTypes.makeEnumType('VerbosityOptions', 'VerbosityOptionsType',
+                                                ['silent', 'quiet', 'all', 'debug'])
+    desc_verbosity_options = r"""determines the level of verbosity for the outer/inner RAVEN runs. \default{all}.
+                             If ``silent'' only errors are displayed.
+                             If ``quiet'' errors and warnings are displayed.
+                             If ``all'' errors, warnings, and messages are displayed.
+                             If ``debug'' errors, warnings, messages, and debug messages are displayed."""
+    input_specs.addSub(InputData.parameterInputFactory('verbosity', contentType=verbosity_options,
+                                                       strictMode=True, descr=desc_verbosity_options))
```
Comment (PaulTalbot-INL) [inline src/Cases.py:80]: someday I bet we want to specify inner/outer/TEAL verbosity here as well, but this is a good start.

File: src/Cases.py (hunk 1 above)
Comment (dylanjm) [reply src/Cases.py:78]: Yes, I can add that in.

File: src/Cases.py (hunk 5 above)
Comment (dylanjm) [reply src/Cases.py:80]: Yes and even HERON verbosity as well. I realized a bit into coding this that we are dealing with verbosity of several different portions of code. I think HERON verbosity could be handled as a flag to the `heron` executable.

**Comment by PaulTalbot-INL** [comment]
I think this is in a good enough state to merge, and the checklists pass. We can address more fine-tune verbosity changes in the future. Merging.

**Review by PaulTalbot-INL — APPROVED** [review]

**Pull request merged**